#163360804 users should be able to Update/Delete an article

jillo-abdullahi opened the pull request:
**What does this PR do?**
-  Implements user ability to update and delete articles

**Description of tasks to be completed?**
- Authenticated users can edit their own article
- Authenticated users who are article owners can choose to delete their article(s)


**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Switch to the appropriate branch with `ft-update-delete-article-163360804`
- Run `npm install` to update your packages.
- Run tests with `npm test`
- Fire up the server with `npm start`
- Navigate to `http://localhost:3000/create_article` to test on Postman.

**What are the relevant pivotal tracker stories?**
- [#163360804](https://www.pivotaltracker.com/story/show/163360804)

**Checklist (edited)**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

jillo-abdullahi committed:
WIP

jillo-abdullahi committed:
Another fix

jillo-abdullahi committed:
Fixup rendering

jillo-abdullahi committed:
feat(create-article): create article

  - implement article creation.
  - integrate with the rest of the app.

[Delivers #1619670921]

jillo-abdullahi committed:
feat(update-delete-article): users can updated or delete an article

   - implement article deletion and updating

[Delivers #163360804]

justmesam commented:
I noticed this has work that is in the other PR, we should wait for that to be merged or it will start having merge conflicts.

bettblake08 commented on src/assets/style/articles/base/_base.scss:
```diff
@@ -0,0 +1,14 @@
+    font-family: Helvetica, Arial, sans-serif;
+    font-size: $m-size;
+}
+button {
```
Change the button css to sass structure.

jillo-abdullahi replied:
On it!

bettblake08 commented on src/assets/style/articles/base/_settings.scss:
```diff
@@ -0,0 +1,14 @@
+$s-size: 1.2rem;
+$m-size: 1.6rem;
+$l-size: 3.2rem;
+$xl-size: 4.8rem;
```
Add a line

jillo-abdullahi replied:
On it!

bettblake08 commented on src/components/views/articles/SingleArticle.js:
```diff
@@ -0,0 +1,235 @@
+/* eslint-disable camelcase */
```
Please place the exceptions inline with the code you are trying to exempt.

jillo-abdullahi replied:
This has been done. Thanks @bettblake08

jillo-abdullahi committed:
feat(update-delete-article): users can updated or delete an article

  - implement article deletion and updating
  - fix an issue with only tags getting updated

[Delivers #163360804]

jillo-abdullahi committed:
Fix an issue with styles folder

jillo-abdullahi committed:
feat(article) user can update and delete article

 - add tests for article update and delete
 - implement feedback.

[Delivers #163360804]

Darius-Ndubi reviewed (approved):
👍

bettblake08 reviewed (approved):
👍

Bryan-Cee reviewed (approved):
👍 LGTM

denomash reviewed (approved):
👍

justmesam commented:
LGTM
